src/main/java/com/hackathon03/codeguru/utils/IncorrectStringEquality.java

```diff
@@ -0,0 +1,18 @@
+package com.hackathon03.codeguru.utils;
+
+public class IncorrectStringEquality {
+
+    public void stringEqualityCheckNoncompliant(String str1, String str2) {
+        // Noncompliant: the == operator doesn't compare the contents of the strings.
+        if(str1 == str2) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.